src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -331,12 +331,27 @@ public static void ApartmentState_NoAttributePresent_STA_Unix()
         [Fact]
         [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
         [SkipOnTargetFramework(TargetFrameworkMonikers.Mono)]
-        public static void CurrentCultureTest_SkipOnDesktopFramework()
+        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)]
+        public static void CurrentCultureTest_DifferentThread()
         {
-            // Cannot access culture properties on a thread object from a different thread
-            var t = new Thread(() => { });
+            CultureInfo culture = (CultureInfo)CultureInfo.CurrentCulture.Clone();
+            CultureInfo uiCulture = (CultureInfo)CultureInfo.CurrentCulture.Clone();
+            var t = new Thread(() => {
+                Assert.Same(culture, Thread.CurrentThread.CurrentCulture);
+                Assert.Same(uiCulture, Thread.CurrentThread.CurrentUICulture);
+            });
+            t.CurrentCulture = culture;
+            t.CurrentUICulture = uiCulture;
+            Assert.Throws<InvalidOperationException>(() => t.CurrentCulture);
+            Assert.Throws<InvalidOperationException>(() => t.CurrentUICulture);
+
+            t.Start();
+            // Cannot access culture properties on a thread object from a different thread once the thread is started
```

Comment:
Nor before it is started apparently, given your asserts above

Reply:
It seems wrong for set to be more lenient than get

Reply:
This is feedback-driven compat quirk, so I do not feel too bad about that the getter does not work. If somebody reports a problem with the getter, we can look into fixing the getter.

We would need to make `CultureInfo.UserDefaultUICulture` public to make the getter work (or use some alternative way to fix this ... reflection, move to CoreLib, ...).